src/client.ts

```diff
@@ -36,98 +44,52 @@ import { Logger, LoggerConfig, makeLogger, defaultLogger } from './logger/logger
  * })
  */
 export class SpvWalletClient {
-  serverUrl: string
-  options: ClientOptions
-  adminKey: bsv.HDPrivateKey | null
-  logger: Logger
- 
+  logger: Logger;
+  http: Requester;
+
+  private xPriv?: bsv.HDPrivateKey;
 
   constructor(serverUrl: string, options: ClientOptions, loggerConfig: LoggerConfig = defaultLogger) {
-    this.serverUrl = serverUrl
-    this.adminKey = null
-    this.logger = makeLogger(loggerConfig)
-
-    if (options.xPriv) {
-      options.xPrivString = options.xPriv.toString()
-      options.xPub = options.xPriv.hdPublicKey
-      options.xPubString = options.xPub.toString()
-    } else if (options.xPrivString) {
-      options.xPriv = bsv.HDPrivateKey.fromString(options.xPrivString)
-      options.xPub = options.xPriv?.hdPublicKey
-      options.xPubString = options.xPub?.toString()
-    } else if (options.xPub) {
-      options.xPriv = undefined
-      options.xPrivString = undefined
-      options.xPubString = options.xPub.toString()
-    } else if (options.xPubString) {
-      options.xPriv = undefined
-      options.xPrivString = undefined
-      options.xPub = bsv.HDPublicKey.fromString(options.xPubString)
-    } else if (options.accessKey) {
-      options.xPriv = undefined
-      options.xPrivString = undefined
-      options.xPub = undefined
-      options.xPubString = undefined
-      options.accessKeyString = options.accessKey.toString()
+    this.logger = makeLogger(loggerConfig);
+
+    if (options.signRequest && options.xPriv) {
+      this.xPriv = bsv.HDPrivateKey.fromString(options.xPriv);
     }
+    this.http = this.initRequester(options, serverUrl);
+  }
 
-    this.options = options
+  private initRequester(options: ClientOptions, serverUrl: string): Requester {
+    if (!options.signRequest && options.xPub) {
+      this.logger.info('Using XPub. Admin requests and SendToRecipients function will not be available.');
+      return Requester.CreateXPubRequester(this.logger, serverUrl, options.xPub);
+    }
 
-    if (options.adminKey) {
-      this.SetAdminKey(options.adminKey)
+    if (!options.signRequest) {
+      throw new ErrorWithDisabledSignRequest(this.logger, options);
     }
```

Comment:
Actually if you choose disabled signRequests, you still could use xPriv as key and then we can calculate xpub from it.

Reply:
New `ClientOptions` has been applied - as we discussed 😄